core/meta/src/TClass.cxx

```diff
@@ -6004,6 +6004,45 @@ void TClass::PostLoadCheck()
          }
       }
    }
+   if (fCollectionProxy) {
+      // Update the related pair's TClass if it has already been created.
+      size_t noffset = 0;
+      if (strncmp(GetName(), "map<", 4) == 0)
+         noffset = 3;
+      else if (strncmp(GetName(), "multimap<", 9) == 0)
+         noffset = 8;
+      else if (strncmp(GetName(), "unordered_map<", 14) == 0)
+         noffset = 13;
+      else if (strncmp(GetName(), "unordered_multimap<", 19) == 0)
+         noffset = 18;
+      if (noffset) {
+         std::string pairname("pair");
+         pairname.append(GetName() + noffset);
+         auto pcl = TClass::GetClass(pairname.c_str(), false, false);
+         if ( pcl && !pcl->IsLoaded() && !pcl->IsSyntheticPair() )
+         {
+            TInterpreter::SuspendAutoLoadingRAII autoloadOff(gInterpreter);
+
+            fCollectionProxy->Reset();
+            TIter nextClass(gROOT->GetListOfClasses());
+            while (auto acl = (TClass*)nextClass()) {
+               if (acl == this) continue;
+               if (acl->fCollectionProxy && acl->fCollectionProxy->GetValueClass() == pcl) {
+                  acl->fCollectionProxy->Reset();
+               }
+            }
+
+            TIter next(pcl->GetStreamerInfos());
```

Comment:
Would it not make sense to update this before telling all collection proxies to update their value class, just in case in the future we get a dependency of the proxies on the value steamer info?

Reply:
Possibly, this is a 'high' risk change (potential order of deletions and/or state change issues, etc.).  So I will try it in a different PR (and there is a potential optimization with the same type of risk 5 lines above that I'll try too)

Reply:
Actually it is confirmed.  The objects (ActionSequence) deleted in the CollectionProxy's Reset are pointing/using the information from the StreamerInfo (that is deleted/destroyed by the StreamerInfo's Clear).